cloudformation/templates/gridss_worker_instance/worker.yaml

```diff
@@ -0,0 +1,221 @@
+            su - "ec2-user" -c 'docker pull 843407916570.dkr.ecr.ap-southeast-2.amazonaws.com/gridss-purple-linx:2.7.3'
+
+  workerInstanceProfile:
+    Type: 'AWS::IAM::InstanceProfile'
+    Properties:
+      Path: /worker/
+      Roles:
+        - !Ref workerInstanceRole
+  workerInstancePolicy:
+    Type: 'AWS::IAM::Policy'
+    Properties:
+      PolicyName: !Sub '${AWS::StackName}_instance_policy'
+      PolicyDocument:
+        Statement:
+          - Effect: Allow
+            Action:
+              - 'ssm:DescribeAssociation'
+              - 'ssm:GetDeployablePatchSnapshotForInstance'
+              - 'ssm:GetDocument'
+              - 'ssm:DescribeDocument'
+              - 'ssm:GetManifest'
+              - 'ssm:GetParameters'
+              - 'ssm:ListAssociations'
+              - 'ssm:ListInstanceAssociations'
+              - 'ssm:PutInventory'
+              - 'ssm:PutComplianceItems'
+              - 'ssm:PutConfigurePackageResult'
+              - 'ssm:UpdateAssociationStatus'
+              - 'ssm:UpdateInstanceAssociationStatus'
+              - 'ssm:UpdateInstanceInformation'
+            Resource: '*'
+          - Effect: Allow
+            Action:
+              - 'ssmmessages:CreateControlChannel'
+              - 'ssmmessages:CreateDataChannel'
+              - 'ssmmessages:OpenControlChannel'
+              - 'ssmmessages:OpenDataChannel'
+            Resource: '*'
+          - Effect: Allow
+            Action:
+              - 'ec2messages:AcknowledgeMessage'
+              - 'ec2messages:DeleteMessage'
+              - 'ec2messages:FailMessage'
+              - 'ec2messages:GetEndpoint'
+              - 'ec2messages:GetMessages'
+              - 'ec2messages:SendReply'
+            Resource: '*'
+          - Effect: Allow
+            Action:
+              - 'cloudwatch:PutMetricData'
+            Resource: '*'
+          - Effect: Allow
+            Action:
+              - 'ec2:DescribeInstanceStatus'
+              - 'ec2:CreateTags'
+            Resource: '*'
+          - Effect: Allow
+            Action:
+              - 'logs:CreateLogGroup'
+              - 'logs:CreateLogStream'
+              - 'logs:DescribeLogGroups'
+              - 'logs:DescribeLogStreams'
+              - 'logs:PutLogEvents'
+            Resource: '*'
+          - Effect: Allow
+            Action:
+              - 'ec2messages:SendReply'
+            Resource: '*'
+      Roles:
+        - !Ref workerInstanceRole
+  workerInstanceRole:
+    Type: 'AWS::IAM::Role'
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - ec2.amazonaws.com
+            Action:
+              - 'sts:AssumeRole'
+      Path: /worker/
+      ManagedPolicyArns:
+        - 'arn:aws:iam::aws:policy/AmazonS3ReadOnlyAccess'
+        - 'arn:aws:iam::843407916570:policy/alexis_mnt_test_write_policy'
```

Comment:
Is there a managed policy that could potentially substitute this custom policy?

Reply:
I needed this initially to add the bwa indices to the ref data (and then re-sync via aws s3) to the bucket and this policy gave me write access to that. I don't think it needs to be there now, but if a user wants to save the data to any bucket they'll need to add a policy to the running instance. Do we have a 'dump' bucket we can code into a policy or are users expected to add in their own policies retrospectively once they know where they want their outputs to go?

Reply:
We have a (per-app) reference data bucket on dev for BWA indices and a bunch of others (`umccr-refdata-dev` or similar)... ideally there should be as few custom policies as possible and those type of resources should be shared if they are heavyweight like refdata. You are right with output files, I can't think of any other way to define and restrict them on a case by case basis.

Reply:
This was a specific hg19 reference as mentioned here https://github.com/hartwigmedical/gridss-purple-linx#reference-data but I'll note that bucket for future reference